src/classifier/llm.py

```diff
@@ -1,312 +1,183 @@
+            concept_description=self.concept.to_markdown()
         )
-        return text.translate(normalize_translation)
 
-    def generate_prompt(self, text_dicts: dict[str, dict[str, str]]) -> str:
-        """Generate the prompt for the LLM."""
-        input_xml = "\n\t".join(
-            f'<passage id="{identifier}">{text_dict["sanitised"]}</passage>'
-            for identifier, text_dict in text_dicts.items()
+        self.agent = Agent(
+            self.model_name,
+            system_prompt=self.system_prompt,
+            result_type=LLMResponse,
         )
 
-        prompt = self.prompt_template.format(
-            PREFERRED_LABEL=self.concept.preferred_label,
-            CONCEPT_DESCRIPTION=self.concept_description,
-            EXAMPLES=self.examples,
-            INPUT_PASSAGES=input_xml,
+    def __getstate__(self):
+        """Handle pickling by removing the unpickleable agent instance."""
+        state = self.__dict__.copy()
+        # Remove the agent instance as it contains unpickleable components
+        del state["agent"]
+        return state
+
+    def __setstate__(self, state):
+        """Recreate the agent instance when unpickling."""
+        self.__dict__.update(state)
+        # Recreate the agent instance
+        self.agent = Agent(
+            self.model_name,
+            system_prompt=self.system_prompt,
+            result_type=LLMResponse,
         )
 
-        # remove all superfluous whitespace from the lines of the prompt
-        prompt = "\n".join(line.strip() for line in prompt.split("\n"))
-
-        return prompt
+    def _validate_response(self, input_text: str, response: LLMResponse) -> None:
+        """Make sure the output text does not augment the input text in unexpected ways"""
+        input_sanitised = LabelledPassage.sanitise(input_text)
+        output_sanitised = LabelledPassage.sanitise(
+            response.marked_up_text.replace("<concept>", "").replace("</concept>", "")
+        )
+        if input_sanitised != output_sanitised:
+            raise LLMOutputMismatchError(
+                input_text=input_sanitised, output_text=output_sanitised
+            )
 
-    def generate_text_dicts(self, texts: list[str]) -> dict[str, dict[str, str]]:
-        """Generate the text dictionary for a batch of texts."""
-        return {
-            generate_identifier(text): {
-                "original": text,
-                "sanitised": self._sanitise_text_for_xml(text),
-            }
-            for text in texts
-        }
+    def predict(self, text: str) -> list[Span]:
+        """Predict whether the supplied text contains an instance of the concept."""
+        response: AgentRunResult[LLMResponse] = self.agent.run_sync(text)
+        self._validate_response(input_text=text, response=response)
+        return Span.from_xml(
+            xml=response.data.marked_up_text,
+            concept_id=self.concept.wikibase_id,
+            labellers=[str(self)],
```

Comment:
it looks like this classifier inherits its `__repr__` method from parent `Classifier`. Given this classifier's behaviour is also dependent on model and system prompt choice, is it worth giving `LLMClassifier` a separate `__repr__` which is sensitive to changes in these?

Reply:
v good point! will add a `__repr__`